pkg/apis/targets/v1alpha1/azure_common_types.go

```diff
@@ -0,0 +1,407 @@
+/*
```

Comment:
It would be nice to also copy the tests in `azure_common_types_test.go` (since this code was copied from sources), because it's quite essential to ensure the deserialization from string (in Kubernetes) to structs (in this Go code) is working as intended.
By having tests, we can more confidently change this code in the future.